### PR TITLE
Feature/ub bv 731

### PR DESCRIPTION
**Pull request by ubozov**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,6 +27,11 @@ git clone https://github.com/Privatix/dapp-installer.git
 cd dapp-installer
 
 go get github.com/spf13/cobra
+go get github.com/josephspurrier/goversioninfo/cmd/goversioninfo
```
[the remark this answers is not in the document]
Comment (ubozov) [reply README.md]: had amended

File: util/misc_windows.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+	c := h.MustFindProc("GetDiskFreeSpaceExW")
+	lpFreeBytesAvailable := uint64(0)
+	c.Call(uintptr(unsafe.Pointer(syscall.StringToUTF16Ptr("\\"))),
+		uintptr(unsafe.Pointer(&lpFreeBytesAvailable)))
+
+	return lpFreeBytesAvailable >= MinAvailableDiskSize
+}
+
+func checkMemory() bool {
+	kernel32, err := syscall.LoadDLL("kernel32.dll")
```
[the remark this answers is not in the document]
Comment (ubozov) [reply util/misc_windows.go]: thanks, I will be use this package, I had amended

File: util/misc_windows.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+	"runtime"
+	"strconv"
+	"strings"
+	"syscall"
+	"unsafe"
+
+	"golang.org/x/sys/windows/registry"
+)
+
+// MinWindowsVersion is supported min windows version (Win7andNewer)
```
[the remark this answers is not in the document]
Comment (ubozov) [reply util/misc_windows.go]: fixed

File: util/misc.go [hunk 4]
```diff
@@ -5,12 +5,12 @@ import (
...
 
-// Recursivly remove all data from directory
-func ClearDirectory(path string) {
-	if err := os.RemoveAll(path); err != nil {
-		panic(err)
-	}
-}
+const (
+	// MinAvailableDiskSize - available min 500MB
+	MinAvailableDiskSize uint64 = 500 * 1024 << 10
```
[the remark this answers is not in the document]
Comment (ubozov) [reply util/misc.go]: fixed

File: command/root.go [hunk 5]
```diff
@@ -14,7 +14,20 @@ type Config struct {
 	Log string
 }
 
-func initialize() *cobra.Command {
+func createRootCmd() *cobra.Command {
+	rootCmd := &cobra.Command{
```
[the remark this answers is not in the document]
Comment (ubozov) [reply command/root.go]: I think this tool more useful, easy configured and usage

File: command/install.go [hunk 6]
```diff
@@ -20,5 +21,13 @@ func createInstallCmd() *cobra.Command {
 func install(ccmd *cobra.Command, args []string) {
 	log.Println("Start install process")
 	fmt.Println("I will be runnning installation process", args)
+
+	if !util.CheckSystemPrerequisites() {
+		log.Println("Installation process was interrupted")
```
[the remark this answers is not in the document]
Comment (ubozov) [reply command/install.go]: What it is mean home-brew logger?

File: command/root.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (ababo) [reply command/root.go]: Sorry for misspelling, meant home-brewed. I.e. the logger located in `util/log`.

File: command/install.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (ababo) [reply command/install.go]: Sorry for misspelling, meant home-brewed. I.e. the logger located in util/log.

[the remark this answers is not in the document]
Comment (ubozov) [reply command/install.go]: ok, I will see how it's used

File: README.md [hunk 7]
```diff
@@ -27,8 +27,19 @@ git clone https://github.com/Privatix/dapp-installer.git
...
 go get github.com/spf13/cobra
+
 ```
 
+for the Windows platform, you also need to perform:
+
+```bash
+go get github.com/josephspurrier/goversioninfo/cmd/goversioninfo
+go get github.com/lxn/win
+go get golang.org/x/sys/windows/registry
```
[the remark this answers is not in the document]
Comment (ubozov) [reply README.md]: fixed

File: command/install.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (ubozov) [reply command/install.go]: I have used your logger

[the remark this answers is not in the document]
Comment (ababo) [reply command/install.go]: It seems you used a wrong one, i.e. not from `util/log`.

[the remark this answers is not in the document]
Comment (ubozov) [reply command/install.go]: I used `util.Logger` type

[the remark this answers is not in the document]
Comment (ababo) [reply command/install.go]: It's going to be removed soon, this is our old logger which isn't good enough.

[the remark this answers is not in the document]
Comment (ubozov) [reply command/install.go]: fixed with using new logger version

File: README.md [hunk 8]
```diff
@@ -26,9 +26,18 @@ Build `dapp-installer` package:
 git clone https://github.com/Privatix/dapp-installer.git
 cd dapp-installer
 
-go get github.com/spf13/cobra
+go get -d github.com/Privatix/dapp-installer
+
 ```
 
+for the Windows platform, you also need to perform:
```
Comment (ababo) [inline README.md]: Make a proper header please.

File: command/install.go [hunk 9]
```diff
@@ -2,23 +2,29 @@ package command
...
 	return &cobra.Command{
 		Use:   "install",
 		Short: "Install dapp core to host",
 		Long:  "Run installation process of dapp core to localhost",
 
-		Run: install,
-	}
-}
+		Run: func(ccmd *cobra.Command, args []string) {
+			log.Info("Start install process")
```
Comment (ababo) [inline command/install.go]: Logger messages should be in lowercase.

File: command/install.go [hunk 10]
```diff
@@ -2,23 +2,29 @@ package command
...
-	"log"
 
+	"github.com/Privatix/dappctrl/util/log"
+	"github.com/privatix/dapp-installer/util"
 	"github.com/spf13/cobra"
 )
 
-func createInstallCmd() *cobra.Command {
+func createInstallCmd(log log.Logger) *cobra.Command {
 	return &cobra.Command{
```
Comment (ababo) [inline command/install.go]: @sofabeat I don't like that we mix several CLI libraries, are you OK with that?

File: util/misc_windows.go [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+	v := win.GetVersion()
+	major := byte(v)
+	return major >= MinWindowsVersion
+}
+
+func checkStorage() bool {
+	h := syscall.MustLoadDLL("kernel32.dll")
+	c := h.MustFindProc("GetDiskFreeSpaceExW")
+	lpFreeBytesAvailable := uint64(0)
+	c.Call(uintptr(unsafe.Pointer(syscall.StringToUTF16Ptr("\\"))),
```
Comment (ababo) [inline util/misc_windows.go]: Why not to use a native [package](https://golang.org/pkg/unicode/utf16)?

Comment (ubozov) [reply util/misc_windows.go]: fixed, used this package

File: command/install.go (hunk 10 above)
Comment (sofabeat) [reply command/install.go]: @ababo @furkhat @ubozov @dzeckelev 
I think you should discuss it directly. Of course, I prefer single toolset, but team should make decision.

File: command/command.go [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+	cmd.execute(logger.Add("command", cmd.Name))
+}
+
+func processedFlags(printVersion func()) bool {
+	v := flag.Bool("version", false, "Prints current dapp-installer version")
+	h := flag.Bool("help", false, "Display dapp-installer help")
+
+	flag.Parse()
+
+	if *v {
```
Comment (furkhat) [inline command/command.go]: Is it safe? I think it should be `if v != nil`.

File: command/command.go [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+		printVersion()
+		return true
+	}
+	if *h {
```
Comment (furkhat) [inline command/command.go]: `if h != nil`

File: command/command.go (hunk 12 above)
Comment (ubozov) [reply command/command.go]: it is initialize with default value (false)

File: command/command.go (hunk 13 above)
Comment (ubozov) [reply command/command.go]: it is initialize with default value (false)

File: README.md [hunk 14]
```diff
@@ -26,34 +26,40 @@ Build `dapp-installer` package:
...
 cd dapp-installer
 
-go get github.com/spf13/cobra
+go get -d github.com/Privatix/dapp-installer
+
 ```
 
-# Usage
+#### Windows platform
+you also need to perform:
```
[the remark this answers is not in the document]
Comment (ubozov) [reply README.md]: removed

File: README.md [hunk 15]
```diff
@@ -26,34 +26,40 @@ Build `dapp-installer` package:
...
-  -h, --help      help for dapp-installer
-  -v, --version   Display the current version of this CLI
+`dapp-installer` or `dapp-installer --help` will show usage and a list of commands:
 
-Use "dapp-installer [command] --help" for more information about a command.
 ```
+Usage:
+   dapp-installer [command] [flags]
+
+ Available Commands:
```
[the remark this answers is not in the document]
Comment (ubozov) [reply README.md]: yes

**Pull request closed without merging**